### ISSUE TITLE
Hello, I am using vlp-16 with poor results. I almost find that SCclosestHistoryFrameID is not found.

### ISSUE THREAD

**Issue by hunkyu** [irapkaist/SC-LeGO-LOAM#1]
What parameters do I need to adjust?

**Comment by kissb2** [comment]
I think changing SEARCH_RATIO (in Scancontext.h, set in 0.1 default) to 1.0 may help you. This is a trick to boost up the speed. If you set this to 1, the scan context distance tries to align all column-shifts (i.e., brute-force) and guarantees to find NN.

And we note that Scan context works for ground-adjusted cases (e.g., mounted at ground vehicle) so for example, if your platform is a drone, whose pitch motion is aggressive, the scan context detector may not work.

**Comment by hunkyu** [comment]
@kissb2 Thank you very much, it is indeed effective, or some wrong id has been detected. Is my vlp-16 lidar currently affected in the car body coordinate system?

**Comment by kissb2** [comment]
maybe there are two reasons. 

1. 
The current distance threshold for SC is set high, 0.5 (see SC_DIST_THRES in Scancontext.h). Actually 0.1-0.15 is recommended as I checked, but if that conservative case, the number of true-positive factors is small. If the number of TP is small, some specific points of a trajectory may be corrected so maximising the number of TP is required to smooth the whole trajectory. 

So I had a different strategy in this repository. Increasing the threshold could make more both true-positive and false-positive factors (so I expect you felt the wrong id detected). However, a single FP factor may destroy the full map quality so I used the robust kernel (find "Cauchy" in mapOptimization.cpp). In conclusion, the (slightly) increased number of FP factors did not affect the map quality, while the more TP enhancing the map optimization. 

also, in the mapOptimizationl.cpp ICP fitness test is followed after loop ID detection, so if a wrong loop is detected, it can be rejected at the ICP fitness test. so taking some high threshold value is okay.

2.
If your LiDAR scan doesn't have a full (360 deg) fov and you visited a place in a reversed way, the loop is hard to be found. So I can say this repository works well for the loops, where the revisit occurs in a same direction.
